debugger/HaxeServer.hx

```diff
@@ -139,8 +139,15 @@ class HaxeServer
             var peer = socket.peer();
             Sys.println("\nReceived connection from " + peer.host + ".");
 
-            HaxeProtocol.writeServerIdentification(socket.output);
-            HaxeProtocol.readClientIdentification(socket.input);
+            try {
```

Comment:
I don't know about all possible uses but the socket has to be closed in case of protocol mismatch. We can add a rethrow in the catch if you think this should still throw an exception